src/Razor/src/Microsoft.AspNetCore.Razor.Language/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -216,7 +216,7 @@ public static TRoot InsertNodesAfter<TRoot>(this TRoot root, SyntaxNode nodeInLi
                 throw new ArgumentNullException(nameof(node));
             }
 
-            var tokens = node.DescendantNodes().Where(n => n.IsToken).Cast<SyntaxToken>();
+            var tokens = node.DescendantNodesAndSelf().Where(n => n.IsToken).Cast<SyntaxToken>();
```

Comment:
Why this change?

Reply:
This is what should happen. Otherwise this method will return empty if you pass it a `SyntaxToken`.